lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -267,41 +274,36 @@ export default class RecipeImplementation implements RecipeInterface {
             }
             ProcessState.getInstance().addState(PROCESS_STATE.CALLING_SERVICE_IN_GET_HANDSHAKE_INFO);
             let response = await this.querier.sendPostRequest(new NormalisedURLPath("/recipe/handshake"), {});
-            let signingKeyLastUpdated = Date.now();
-            if (this.handshakeInfo !== undefined) {
-                if (
-                    response.jwtSigningPublicKeyExpiryTime === this.handshakeInfo.jwtSigningPublicKeyExpiryTime &&
-                    response.jwtSigningPublicKey === this.handshakeInfo.jwtSigningPublicKey
-                ) {
-                    signingKeyLastUpdated = this.handshakeInfo.signingKeyLastUpdated;
-                }
-            }
+
             this.handshakeInfo = {
-                signingKeyLastUpdated,
-                jwtSigningPublicKey: response.jwtSigningPublicKey,
+                jwtSigningPublicKeyList: response.jwtSigningPublicKeyList,
                 antiCsrf,
                 accessTokenBlacklistingEnabled: response.accessTokenBlacklistingEnabled,
-                jwtSigningPublicKeyExpiryTime: response.jwtSigningPublicKeyExpiryTime,
                 accessTokenValidity: response.accessTokenValidity,
                 refreshTokenValidity: response.refreshTokenValidity,
             };
+
+            this.updateJwtSigningPublicKeyInfo(
+                response.jwtSigningPublicKeyList,
+                response.jwtSigningPublicKey,
+                response.jwtSigningPublicKeyExpiryTime
+            );
+
             if (this.isInServerlessEnv) {
                 storeIntoTempFolderForServerlessCache(SERVERLESS_CACHE_HANDSHAKE_INFO_FILE_PATH, this.handshakeInfo);
             }
```

Comment:
This reminds me that the cached data for serverless can be stored in the older format (without `jwtSigningPublicKeyList`.. need to discuss the implications of this.

Reply:
TODO: Bhumil is testing if we can remove the serverless stuff entirely...